### PR TITLE
fix(auth): Return session expired on revoke token

### PR DESCRIPTION
**Pull request by royjit**
*Issue #, if available:* https://github.com/aws-amplify/amplify-swift/issues/2599

*Description of changes:*

*Check points: (check or cross out if not relevant)*

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

*DataStore checkpoints (check when completed)*

- [ ] Ran AWSDataStorePluginIntegrationTests
- [ ] Ran AWSDataStorePluginV2Tests
- [ ] Ran AWSDataStorePluginMultiAuthTests
- [ ] Ran AWSDataStorePluginCPKTests
- [ ] Ran AWSDataStorePluginAuthCognitoTests
- [ ] Ran AWSDataStorePluginAuthIAMTests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/FetchAuthorizationSession/InformSessionError.swift [hunk 1]
```diff
@@ -37,12 +38,33 @@ struct InformSessionError: Action {
...
+
+
+        if let sdkError = error as? SdkError<GetCredentialsForIdentityOutputError>,
+           case .client(let clientError, _) = sdkError,
+           case .retryError(let retryError) = clientError,
+           let serviceError = retryError as? SdkError<GetCredentialsForIdentityOutputError>,
+           case .service(let internalError, _) = serviceError
+        {
+            return isNotAuthorizedError(internalError)
+        }
```
Comment (atierian) [inline AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/FetchAuthorizationSession/InformSessionError.swift:59]: Does it make sense to create a generic nested function that accepts a type for the `SdkError` generic? That would reduce duplicate code and help reduce the risk of these error extraction implementations drifting apart.

Comment (royjit) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/FetchAuthorizationSession/InformSessionError.swift]: Updated the code to get the underlying error

**Review by atierian — APPROVED** [review]
LGTM

**Comment by ethan021021** [comment]
Can you please release this for V1 as well? Thanks!

**Pull request merged**